blaze-server/src/main/scala/org/http4s/server/blaze/Http2NodeStage.scala

```diff
@@ -187,12 +187,11 @@ private class Http2NodeStage[F[_]](
 
       async.unsafeRunAsync {
         try service(req)
-          .recoverWith(serviceErrorHandler(req).andThen(_.widen[MaybeResponse[F]]))
-          .handleError { _ =>
-            Response[F](InternalServerError, req.httpVersion)
-          }
-          .map(renderResponse(_))
-        catch serviceErrorHandler(req).andThen(_.widen[MaybeResponse[F]])
+          .getOrElse(Response.notFound)
```

Comment:
Not apropos of this commit, but this needs to look more like `Http1ServerStage` to thread shift properly. 😟